lab_4_fill_words_by_ngrams/main.py

```diff
@@ -318,6 +493,13 @@ def __init__(
                 NGramLanguageModel instance to use for text generation
             word_processor (WordProcessor): WordProcessor instance to handle text processing
         """
+        self._generator_type = generator_type
+        if generator_type == 0:
+            self._generator = GreedyTextGenerator(language_model, word_processor)
+        elif generator_type == 1:
+            self._generator = TopPGenerator(language_model, word_processor, 0.5)
+        elif self._generator_type == 2:
+            self._generator = BeamSearchTextGenerator(language_model, word_processor, 5)
```

Comment:
maybe use dictionary here? to avoid situations when it is not initialized due to wrong input

Reply:
fixed